src/types/rule.cpp

```diff
@@ -165,19 +207,12 @@ const Expression::Ptr& Rule::getCondition() const
  *
  * @return Tags.
  */
-std::vector<std::string>& Rule::getTags()
-{
-	return _tags;
-}
-
-/**
- * Returns the tags of the YARA rule.
- *
- * @return Tags.
- */
-const std::vector<std::string>& Rule::getTags() const
+std::vector<std::string> Rule::getTags() const
 {
-	return _tags;
+	std::vector<std::string> output;
```

Comment:
If you know the size of `std::vector` beforehand then reserve the memory for it so you don't perform more allocations than necessary.